serf/serf_test.go

```diff
@@ -836,6 +836,68 @@ func TestSerfRemoveFailedNode(t *testing.T) {
 	testMember(t, s3.Members(), s2Config.NodeName, StatusLeft)
 }
 
+func TestSerfRemoveFailedNode_prune(t *testing.T) {
+	s1Config := testConfig()
+	s2Config := testConfig()
+	s3Config := testConfig()
+
+	s1, err := Create(s1Config)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	s2, err := Create(s2Config)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	s3, err := Create(s3Config)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	defer s1.Shutdown()
```

Comment:
You should interlace these after the if/err checks above. That way if `Create` for s3 fails you don't leak the running server for s1 and s2.

Reply:
I updated this, can you check to see if what I did matched what you wanted?

Reply:
:+1: